model/manifest.go

```diff
@@ -13,13 +13,41 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+const (
+	PLUGIN_CONFIG_TYPE_TEXT      = "text"
+	PLUGIN_CONFIG_TYPE_BOOL      = "bool"
+	PLUGIN_CONFIG_TYPE_RADIO     = "radio"
+	PLUGIN_CONFIG_TYPE_DROPDOWN  = "dropdown"
+	PLUGIN_CONFIG_TYPE_GENERATED = "generated"
+)
+
+type PluginOption struct {
+	DisplayName string `json:"display_name" yaml:"display_name"`
+	Value       string `json:"value" yaml:"value"`
+}
+
+type PluginUISetting struct {
+	DisplayName string          `json:"display_name" yaml:"display_name"`
+	Type        string          `json:"type" yaml:"type"`
+	HelpText    string          `json:"help_text" yaml:"help_text"`
+	Default     interface{}     `json:"default" yaml:"default"`
+	Options     []*PluginOption `json:"options,omitempty" yaml:"options,omitempty"`
+}
+
+type PluginUISettings struct {
+	HeaderText string                      `json:"header_text" yaml:"header_text"`
+	FooterText string                      `json:"footer_text" yaml:"footer_text"`
+	Settings   map[string]*PluginUISetting `json:"settings" yaml:"settings"`
+}
+
 type Manifest struct {
-	Id          string           `json:"id" yaml:"id"`
-	Name        string           `json:"name,omitempty" yaml:"name,omitempty"`
-	Description string           `json:"description,omitempty" yaml:"description,omitempty"`
-	Version     string           `json:"version" yaml:"version"`
-	Backend     *ManifestBackend `json:"backend,omitempty" yaml:"backend,omitempty"`
-	Webapp      *ManifestWebapp  `json:"webapp,omitempty" yaml:"webapp,omitempty"`
+	Id          string            `json:"id" yaml:"id"`
+	Name        string            `json:"name,omitempty" yaml:"name,omitempty"`
+	Description string            `json:"description,omitempty" yaml:"description,omitempty"`
+	Version     string            `json:"version" yaml:"version"`
+	Backend     *ManifestBackend  `json:"backend,omitempty" yaml:"backend,omitempty"`
+	Webapp      *ManifestWebapp   `json:"webapp,omitempty" yaml:"webapp,omitempty"`
+	UISettings  *PluginUISettings `json:"ui_settings,omitempty" yaml:"ui_settings,omitempty"`
```

Comment:
"ui_settings" reads to me like "settings that change the way the user interface behaves". "settings_schema" or just "settings" would be better imo.

E.g.:

```yaml
settings:
    header_text: theheadertext
    footer_text: thefootertext
    fields:
        thesetting:
            display_name: thedisplayname
            type: text
```

Reply:
I'm also tempted to suggest changing "header_text" and "footer_text" to simply `header` and `footer`. I don't think the "_text" suffix is really necessary.

Reply:
I don't like just "settings" because it implies you need to set this to have any settings for your plugin when in reality it's only if you want them exposed in the system console. I'll go with "settings_schema"

0/5 on header/footer, I'll update those as well